hs_file_types/models/base.py

```diff
@@ -1409,6 +1377,15 @@ def _xml_file_short_path(self, resmap=True):
             xml_file_name = os.path.join(file_folder, xml_file_name)
         return xml_file_name
 
+    def read_metadata_file(self):
+        istorage = self.resource.get_irods_storage()
+        with istorage.open(self.metadata_file_path, mode='r') as f:
+            return f.read()
```

Comment:
istorage.download(self.metadata_file_path).read() should suffice here.

Reply:
I'll try that out